### PR TITLE
feat(Accounts): Add Condition Field In Pricing Rule

### PR DESCRIPTION
**Pull request by abhishekbalam**
Condition field to add conditional expressions like in [Assignment Rule](https://docs.erpnext.com/docs/user/manual/en/automation/assignment-rule):
`territory == 'India' and order_type != 'Maintenance'`

image

All fields available in the transaction (like Sales Invoice, Sales Order etc) are available
as their fieldnames directly for use in the conditions.

### CODE REVIEW

**Comment by abhishekbalam** [comment]
@rohitwaghchaure @deepeshgarg007 

Docs Link:
https://github.com/frappe/erpnext_documentation/pull/130

File: erpnext/accounts/doctype/pricing_rule/pricing_rule.json [hunk 1]
```diff
@@ -12,6 +13,7 @@
   "apply_on",
   "price_or_product_discount",
   "warehouse",
+  "condition",
```
Comment (rohitwaghchaure) [inline erpnext/accounts/doctype/pricing_rule/pricing_rule.json]: should be in a different section

File: erpnext/accounts/doctype/pricing_rule/utils.py [hunk 2]
```diff
@@ -51,6 +53,24 @@ def get_pricing_rules(args, doc=None):
...
+def filter_pricing_rule_based_on_condition(pricing_rules, doc=None):
+	filtered_pricing_rules = []
+	if doc:
+		for pricing_rule in pricing_rules:
+			if pricing_rule.condition:
+				try:
+					if frappe.safe_eval(pricing_rule.condition, None, doc.as_dict()):
+						filtered_pricing_rules.append(pricing_rule)
+				except Exception as e:
+					frappe.msgprint(_("Pricing Rule - " + pricing_rule.name + " - <b>condition</b> field error:<br>" + \
```
Comment (rohitwaghchaure) [inline erpnext/accounts/doctype/pricing_rule/utils.py]: No need of msgprint, if condition is not satisfying then don't append

File: erpnext/accounts/doctype/pricing_rule/pricing_rule.json (hunk 1 above)
Comment (abhishekbalam) [reply erpnext/accounts/doctype/pricing_rule/pricing_rule.json]: Moved to "Advanced Settings" section.
Is that fine?

**Comment by abhishekbalam** [comment]
Will write and push tests today itself.

**Comment by deepeshgarg007** [comment]
@abhishekbalam any updates on this?

**Comment by abhishekbalam** [comment]
@deepeshgarg007 
Will do this asap, I'm having bandwidth issues.

**Comment by abhishekbalam** [comment]
@rohitwaghchaure @deepeshgarg007 

Test for condition field added.

image

**Comment by abhishekbalam** [comment]
Hi, requesting review since this PR is open for too long:
@nabinhait @rohitwaghchaure @deepeshgarg007

**Review by deepeshgarg007 — APPROVED** [review]

**Pull request merged**